src/hb-mutex.hh

```diff
@@ -48,6 +48,17 @@
 /* Defined externally, i.e. in config.h; must have typedef'ed hb_mutex_impl_t as well. */
 
 
+#elif !defined(HB_NO_MT) && defined(HB_MUTEX_IMPL_STD_MUTEX)
```

Comment:
No need for extra symbol. We should just use this as backup.  When we added `std::atomic` to `hb-atomic.hh`, I removed all other implementations except for the compiler builtins.

The only reason to keep pthreads and win32 impls above this is that `std::mutex` probably requires linking to libstdc++; right?  We're moving towards that anyway; and trying to use this instead of erring sounds better to me.